packages/shared/src/features/shortcuts/components/modals/CustomLinksModal.tsx

```diff
@@ -0,0 +1,179 @@
+import type { ReactElement } from 'react';
+import React, { useEffect, useRef } from 'react';
+import {
+  Button,
+  ButtonColor,
+  ButtonSize,
+  ButtonVariant,
+} from '../../../../components/buttons/Button';
+import { UserIcon, SitesIcon } from '../../../../components/icons';
+import type { ModalProps } from '../../../../components/modals/common/Modal';
+import { Modal } from '../../../../components/modals/common/Modal';
+import { IconSize } from '../../../../components/Icon';
+import { CardSelection } from '../CardSelection';
+import { LinksForm } from '../LinksForm';
+import { useShortcutLinks } from '../../hooks/useShortcutLinks';
+import { useShortcuts } from '../../contexts/ShortcutsProvider';
+import { useLogContext } from '../../../../contexts/LogContext';
+import { LogEvent, TargetType } from '../../../../lib/log';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../../../../components/typography/Typography';
+import { HorizontalSeparator } from '../../../../components/utilities';
+import { Switch } from '../../../../components/fields/Switch';
+import { useSettingsContext } from '../../../../contexts/SettingsContext';
+
+export default function CustomLinksModal(props: ModalProps): ReactElement {
+  const { logEvent } = useLogContext();
+  const { showTopSites, toggleShowTopSites } = useSettingsContext();
+  const { onSaveChanges, formRef, hasTopSites } = useShortcutLinks();
+  const { isManual, setIsManual, onRevokePermission, setShowPermissionsModal } =
+    useShortcuts();
+
+  const logRef = useRef<typeof logEvent>();
+  logRef.current = logEvent;
+
+  const displayRevoke = !isManual && hasTopSites !== null;
+
+  useEffect(() => {
+    document.body.classList.add('hidden-scrollbar');
```

Comment:
Maybe this is actually something we could implement in our modals in general?
Would love to here from web guild, could you potentially take it forward?

Can be out of scope of this PR ofcourse.

Reply:
iirc, it was possible to do it in our `base.css` - the only problem was from the Post modal where it produces some layout shift, or whatever. Can't exactly remember, but there was something that held us back from doing it there in general.